src/Symfony/Component/Messenger/Tests/WorkerTest.php

```diff
@@ -245,6 +247,37 @@ public function testWorkerWithMultipleReceivers()
         $this->assertSame([$envelope1, $envelope2, $envelope3, $envelope4, $envelope5, $envelope6], $processedEnvelopes);
     }
 
+    public function testWorkerLimitQueues()
+    {
+        $receiver1 = $this->createMock(QueueReceiverInterface::class);
+        $receiver1->expects($this->once())
+            ->method('getFromQueues')
+            ->with(['foo'])
+            ->willReturn([new Envelope(new DummyMessage('message1'))])
+        ;
+
+        $bus = $this->getMockBuilder(MessageBusInterface::class)->getMock();
+
+        $dispatcher = new EventDispatcher();
+        $dispatcher->addSubscriber(new StopWorkerOnMessageLimitListener(1));
+
+        $worker = new Worker([$receiver1], $bus, $dispatcher);
+        $this->expectException(RuntimeException::class);
```

Comment:
Why should a `RuntimeException` be thrown here?

Reply:
hm, not sure what i was thinking. makes no sense to me now - we have the expected call above, i added that we don't expect plain `get` to be called when we specify the queue names